microcosm/caching.py

```diff
@@ -0,0 +1,91 @@
+"""
+Component caches.
+
+The object graph caches components to avoid repeated factory calls.
+
+Alternate cache implementations can be used to speed up performance
+(or to force re-creation).
+
+"""
+from abc import ABCMeta, abstractmethod
+from six import add_metaclass
+
+
+@add_metaclass(ABCMeta)
+class Cache(object):
+    """
+    A cache uses the basic dictionary interface and defines a name.
+
+    """
+
+    @classmethod
+    def name(self):
+        raise NotImplementedError
+
+    @abstractmethod
+    def __getitem__(self, name):
+        pass
+
+    @abstractmethod
+    def __setitem__(self, name, component):
+        pass
+
+    @abstractmethod
+    def __delitem__(self, name):
+        pass
+
+
+class NaiveCache(dict, Cache):
```

Comment:
It's not a noop though; it is caching. If it weren't, we'd have to create every collaborator every time.